lib/s3/managed_upload.js

```diff
@@ -590,8 +590,18 @@ AWS.S3.ManagedUpload = AWS.util.inherit({
       self.body.resume();
     }
 
+    // cleanup multipartReq listeners
+    if (self.multipartReq) {
```

Comment:
CodeCov is being a tad too extreme on this PR, but I think it would be helpful to add a unit test that ensures listeners are removed and that `isDoneChunking` is left `false`.